receiver/prometheusexecreceiver/factory_test.go

```diff
@@ -78,7 +78,6 @@ func TestCreateTraceAndMetricsReceiver(t *testing.T) {
 				ScrapeConfigs: []*promconfig.ScrapeConfig{
 					{
 						ScrapeInterval:  model.Duration(60 * time.Second),
-						ScrapeTimeout:   model.Duration(10 * time.Second),
```

Comment:
This is failing locally for me:

```
--- FAIL: TestCreateTraceAndMetricsReceiver (0.00s)
    /home/jpkroehling/Projects/src/github.com/open-telemetry/opentelemetry-collector-contrib/receiver/prometheusexecreceiver/factory_test.go:107: 
        	Error Trace:	factory_test.go:107
        	Error:      	Not equal: 
        	            	expected: &prometheusexecreceiver.prometheusExecReceiver{params:component.ReceiverCreateSettings{TelemetrySettings:component.TelemetrySettings{Logger:(*zap.Logger)(0xc0002171a0), TracerProvider:trace.noopTracerProvider{}, MeterProvider:metric.noopMeterProvider{}, MetricsLevel:-1}, BuildInfo:component.BuildInfo{Command:"otelcol", Description:"OpenTelemetry Collector", Version:"latest"}}, config:(*prometheusexecreceiver.Config)(0xc0002160c0), consumer:consumer.Metrics(nil), promReceiverConfig:(*prometheusreceiver.Config)(0xc000b5b490), subprocessConfig:(*subprocessmanager.SubprocessConfig)(0xc000b746f0), port:9104, prometheusReceiver:component.MetricsReceiver(nil), shutdownCh:(chan struct {})(nil)}
        	            	actual  : &prometheusexecreceiver.prometheusExecReceiver{params:component.ReceiverCreateSettings{TelemetrySettings:component.TelemetrySettings{Logger:(*zap.Logger)(0xc000217140), TracerProvider:trace.noopTracerProvider{}, MeterProvider:metric.noopMeterProvider{}, MetricsLevel:-1}, BuildInfo:component.BuildInfo{Command:"otelcol", Description:"OpenTelemetry Collector", Version:"latest"}}, config:(*prometheusexecreceiver.Config)(0xc0002160c0), consumer:consumer.Metrics(nil), promReceiverConfig:(*prometheusreceiver.Config)(0xc000b5b420), subprocessConfig:(*subprocessmanager.SubprocessConfig)(0xc000b74600), port:9104, prometheusReceiver:component.MetricsReceiver(nil), shutdownCh:(chan struct {})(nil)}
        	            	
        	            	Diff:
        	            	--- Expected
        	            	+++ Actual
        	            	@@ -75,3 +75,3 @@
        	            	      ScrapeInterval: (model.Duration) 60000000000,
        	            	-     ScrapeTimeout: (model.Duration) 0,
        	            	+     ScrapeTimeout: (model.Duration) 10000000000,
        	            	      MetricsPath: (string) (len=8) "/metrics",
        	Test:       	TestCreateTraceAndMetricsReceiver
FAIL
FAIL	github.com/open-telemetry/opentelemetry-collector-contrib/receiver/prometheusexecreceiver	0.019s
FAIL
```

Reply:
my mad, I right understood now how test is done.

First, it creates `metricReceiver` using the factory so it has the default values

Then it creates `wantPer` manually so if you do not set explicitly the value it will be the golang zero value

Finally, it assert equal on both.

So it seems better to reuse the default values constants to define `wantPer` from this way we are sure both receivers are equal

To fully cover the problematic case I also added a new asserts to be sure the time based default values used in factory are never 0 which are bad value for receiver.

```
LANG=eng make -C ./receiver/prometheusexecreceiver test 
make: Entering directory '/home/qmanfroi/git/signalfx/opentelemetry-collector-contrib/receiver/prometheusexecreceiver'
go test -race -v -timeout 300s --tags="" ./...
=== RUN   TestLoadConfig
--- PASS: TestLoadConfig (0.01s)
=== RUN   TestCreateTraceAndMetricsReceiver
--- PASS: TestCreateTraceAndMetricsReceiver (0.01s)
=== RUN   TestExecKeyMissing
--- PASS: TestExecKeyMissing (0.01s)
=== RUN   TestEndToEnd
--- PASS: TestEndToEnd (11.71s)
=== RUN   TestConfigBuilderFunctions
=== RUN   TestConfigBuilderFunctions/no_command
=== RUN   TestConfigBuilderFunctions/normal_config
=== RUN   TestConfigBuilderFunctions/lots_of_defaults
=== RUN   TestConfigBuilderFunctions/no_command#01
=== RUN   TestConfigBuilderFunctions/normal_config#01
=== RUN   TestConfigBuilderFunctions/lots_of_defaults#01
--- PASS: TestConfigBuilderFunctions (0.00s)
    --- PASS: TestConfigBuilderFunctions/no_command (0.00s)
    --- PASS: TestConfigBuilderFunctions/normal_config (0.00s)
    --- PASS: TestConfigBuilderFunctions/lots_of_defaults (0.00s)
    --- PASS: TestConfigBuilderFunctions/no_command#01 (0.00s)
    --- PASS: TestConfigBuilderFunctions/normal_config#01 (0.00s)
    --- PASS: TestConfigBuilderFunctions/lots_of_defaults#01 (0.00s)
=== RUN   TestFillPortPlaceholders
=== RUN   TestFillPortPlaceholders/port_is_defined_by_user
=== RUN   TestFillPortPlaceholders/no_string_templating
=== RUN   TestFillPortPlaceholders/no_port_defined
--- PASS: TestFillPortPlaceholders (0.00s)
    --- PASS: TestFillPortPlaceholders/port_is_defined_by_user (0.00s)
    --- PASS: TestFillPortPlaceholders/no_string_templating (0.00s)
    --- PASS: TestFillPortPlaceholders/no_port_defined (0.00s)
=== RUN   TestGetDelayAndComputeCrashCount
=== RUN   TestGetDelayAndComputeCrashCount/GetDelay_test
=== RUN   TestGetDelayAndComputeCrashCount/healthy_process_1
=== RUN   TestGetDelayAndComputeCrashCount/healthy_process_2
=== RUN   TestGetDelayAndComputeCrashCount/unhealthy_process_1
=== RUN   TestGetDelayAndComputeCrashCount/unhealthy_process_2
=== RUN   TestGetDelayAndComputeCrashCount/unhealthy_process_3
=== RUN   TestGetDelayAndComputeCrashCount/unhealthy_process_4
--- PASS: TestGetDelayAndComputeCrashCount (0.00s)
    --- PASS: TestGetDelayAndComputeCrashCount/GetDelay_test (0.00s)
    --- PASS: TestGetDelayAndComputeCrashCount/healthy_process_1 (0.00s)
    --- PASS: TestGetDelayAndComputeCrashCount/healthy_process_2 (0.00s)
    --- PASS: TestGetDelayAndComputeCrashCount/unhealthy_process_1 (0.00s)
    --- PASS: TestGetDelayAndComputeCrashCount/unhealthy_process_2 (0.00s)
    --- PASS: TestGetDelayAndComputeCrashCount/unhealthy_process_3 (0.00s)
    --- PASS: TestGetDelayAndComputeCrashCount/unhealthy_process_4 (0.00s)
PASS
ok      github.com/open-telemetry/opentelemetry-collector-contrib/receiver/prometheusexecreceiver       11.866s
=== RUN   TestFormatEnvSlice
=== RUN   TestFormatEnvSlice/empty_slice
=== RUN   TestFormatEnvSlice/one_entry
=== RUN   TestFormatEnvSlice/three_entries
--- PASS: TestFormatEnvSlice (0.00s)
    --- PASS: TestFormatEnvSlice/empty_slice (0.00s)
    --- PASS: TestFormatEnvSlice/one_entry (0.00s)
    --- PASS: TestFormatEnvSlice/three_entries (0.00s)
=== RUN   TestRun
=== RUN   TestRun/sleep_4ms_and_succeeds
=== RUN   TestRun/sleep_4ms_and_fail
{"level":"error","ts":1644254609.092799,"caller":"subprocessmanager/manager.go:101","msg":"subprocess output line","output":"exit status 1","stacktrace":"github.com/open-telemetry/opentelemetry-collector-contrib/receiver/prometheusexecreceiver/subprocessmanager.(*SubprocessConfig).pipeSubprocessOutput\n\t/home/qmanfroi/git/signalfx/opentelemetry-collector-contrib/receiver/prometheusexecreceiver/subprocessmanager/manager.go:101"}
=== RUN   TestRun/proper_parsing_of_arguments_with_spaces,_surrounded_by_quotes
=== RUN   TestRun/sleep_4_and_succeeds,_with_quotes
=== RUN   TestRun/normal_process_2,_normal_process_exit
{"level":"info","ts":1644254609.3431714,"caller":"subprocessmanager/manager.go:99","msg":"subprocess output line","output":"go version go1.17.6 linux/amd64"}
=== RUN   TestRun/shellquote_error
--- PASS: TestRun (0.54s)
    --- PASS: TestRun/sleep_4ms_and_succeeds (0.16s)
    --- PASS: TestRun/sleep_4ms_and_fail (0.13s)
    --- PASS: TestRun/proper_parsing_of_arguments_with_spaces,_surrounded_by_quotes (0.12s)
    --- PASS: TestRun/sleep_4_and_succeeds,_with_quotes (0.12s)
    --- PASS: TestRun/normal_process_2,_normal_process_exit (0.01s)
    --- PASS: TestRun/shellquote_error (0.00s)
PASS
ok      github.com/open-telemetry/opentelemetry-collector-contrib/receiver/prometheusexecreceiver/subprocessmanager     (cached)
make: Leaving directory '/home/qmanfroi/git/signalfx/opentelemetry-collector-contrib/receiver/prometheusexecreceiver'

```

hope it suits you

Reply:
This looks good now: the `want` var has the default value, so, we are ensuring that the default configuration has the default scrape timeout value.